### ISSUE TITLE
jsonio java.lang.reflect.InvocationTargetException error

### ISSUE THREAD

**Issue by adanaee** [wbuchanan/StataJSON#34]
Hi,
I was hoping to get some insight on this issue. 
When I try to run this command:
`jsonio kv, file("C:\Users\andis\OneDrive\Documents\FOTN\output2.json") nourl`

I get the following error:
`java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
        at java.base/java.lang.reflect.Method.invoke(Unknown Source)
        at com.stata.Javacall.load(Javacall.java:130)
        at com.stata.Javacall.load(Javacall.java:90)
Caused by: java.util.NoSuchElementException
        at java.base/java.util.LinkedList.removeFirst(Unknown Source)
        at java.base/java.util.LinkedList.pop(Unknown Source)
        at org.paces.Stata.Input.FlatJSON.checkName(FlatJSON.java:441)
        at org.paces.Stata.Input.FlatJSON.flatten(FlatJSON.java:181)
        at org.paces.Stata.Input.FlatJSON.flatten(FlatJSON.java:121)
        at org.paces.Stata.Input.InJSON.insheetFile(InJSON.java:155)
        ... 6 more
r(5100);`

**Comment by wbuchanan** [comment]
@adanaee 
Are you able to share the file that you were trying to read in?

**Comment by wbuchanan** [comment]
@adanaee 
Also, do you have `brewscheme`, `StataOS`, `StataFileSystem`, and/or any of the other programs I've developed using Java plugins installed on your machine?  This issue https://github.com/wbuchanan/StataJavaUtilities/issues/1 is related and it ended up being a classpath resolution issue.

**Comment by adanaee** [comment]
Sure thing, this is what I am trying to read:
[output2.txt](https://github.com/wbuchanan/StataJSON/files/3704244/output2.txt)

**Comment by adanaee** [comment]
+No, I don't have those program.

**Comment by adanaee** [comment]
@wbuchanan 
Quick update: The error goes away when I remove the square brackets at the start and end of the file. Not sure if this is a bug...

**Comment by emawbgit** [comment]
Hello, I am having the same error message displayed in this thread. It happens when I import a huge number of json (>60 in a loop). I tried to check if the java error occurs due to a faulty structure of the json that I import (though all have the same structure and apparently no extra/missing square brackets). Find attached the json that jsonio is failing to import.

[tryoutput_83.txt](https://github.com/wbuchanan/StataJSON/files/5882443/tryoutput_83.txt)

And this is the code:

*Import all the requests (jsons) as dta
clear all
jsonio kv, file("C:\Users\wb562201\Desktop\apidata\tryoutput_1.json") nourl
gen request=1
**qui save "C:\Users\wb562201\Desktop\apidata\request_1.dta",replace 
forval k=2/301{
	sleep 2000
	clear all
	jsonio kv, file("C:\Users\wb562201\Desktop\apidata\tryoutput_`k'.json") nourl 
	gen request=`k'
	qui save "C:\Users\wb562201\Desktop\apidata\request_`k'.dta",replace 	
	sleep 2000
}** 	 


My educated guess would be that, for some reason, the jsonio works well for a given number of iterations.
Please let me know if you need further clarifications.

Thanks,
Emanuele

**Comment by wbuchanan** [comment]
@emawbgit have you installed the other Java binary mentioned above?

**Comment by emawbgit** [comment]
I had installed **brewscheme**. For StataOS and StataFileSystem not sure what should I do: I had manually installed the **clicmd**.ado, **sysresources**.ado [cfr the StataOS], and **filesys**.ado [cfr. StataFileSystem]. Yet, I keep on getting the error.
Maybe I'm missing something trivial, but do you think I should do something else?

**Comment by wbuchanan** [comment]
@emawbgit 
This program and the others all depend on https://github.com/wbuchanan/StataJavaUtilities.  So all of the helper classes I wrote to access the data and metadata are stored in that binary.  In earlier versions of brewscheme and the other packages, I had compiled that library with the other Java binaries.  The problem that caused was a classpath issue when one program used a different version of the StataJavaUtilities library than another package.  So, I decoupled them.  This way all of the other packages assume that a compiled version of StataJavaUtilities is available on the system and only require a minimum version being available.  The invocation error mentioned above has to do with Stata not being able to locate the classes in StataJavaUtilities.  If that binary (the file that ends in .jar) is available on the ADOPATH, you shouldn't get that specific error.
